torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1038,13 +1041,23 @@ std::shared_ptr<ProcessGroup::Work> ProcessGroupNCCL::collective(
   syncStreams(devices, ncclEvents_[key], ncclStreams_[key]);
 
   // Work itself will create the CUDA events on all GPUs of tensors
-  auto work = initWork(devices, rank_, opType);
+  bool can_profile = outputs.size() == 1;
+  auto work = initWork(devices, rank_, opType, can_profile ? profiling_title : nullptr);
 
   // Store references to outputs and futureNCCLCallbackStream to be used by
   // WorkNCCL::getFuture.
   work->outputs_ = std::make_shared<std::vector<at::Tensor>>(outputs);
   work->futureNCCLCallbackStreams_ = futureNCCLCallbackStreams_;
 
+  if (work->recordFunctionEndCallback_) {
```

Comment:
We probably need to enhance `pointTopoint` as well to cover send and recv?

Reply:
I think it's fine to add as a follow up PR, but can we file a GH issue for this (and any other follow up tasks?)

Reply:
https://github.com/pytorch/pytorch/issues/47482